src/vs/workbench/contrib/debug/browser/debugSession.ts

```diff
@@ -1291,6 +1293,49 @@ export class DebugSession implements IDebugSession, IDisposable {
 		);
 	}
 
+	private enableDependentBreakpoints(thread: Thread) {
+		const frame = thread.getTopStackFrame();
+		if (frame === undefined) {
+			return;
+		}
+
+		if (thread.stoppedDetails && thread.stoppedDetails.reason !== 'breakpoint') {
+			return;
+		}
+
+		// find the current breakpoints
+		const breakpoints = this.model.getBreakpoints({ uri: frame?.source.uri }).filter(bp => {
```

Comment:
The stoppedDetails can optionally include the `hitBreakpoints`. This method could be implemented as `enableDependentBreakpoints(thread: Thread, hitBreakpoints?: string[])`, which is called with the hitBreakpoints if they're included in the details, and only if they aren't included it's called where it currently is without them.

Then this line would be

```suggestion
		const breakpoints = hitBreakpoints.map(someLookupFunction) || this.model.getBreakpoints({ uri: frame?.source.uri }).filter(bp => {
```

This would be a little more reliable (resilient to dirty editors) and faster for DAs that include the hitBreakpoints.